### PR TITLE
sile 0.15.3

### PR DESCRIPTION
**Pull request by alerque**
The [v0.15.0](https://github.com/sile-typesetter/sile/releases/tag/v0.15.0) was a huge one with many breaking changes, dependency changes, and a new CLI in Rust, not Lua. The build system updates were mostly implemented in this formula already because the HEAD option has had them working for a while. This mostly just promotes all those changes to the stable tagged build.

See also upstream releases notes on [v0.15.1](https://github.com/sile-typesetter/sile/releases/tag/v0.15.1), [v0.15.2](https://github.com/sile-typesetter/sile/releases/tag/v0.15.2), and [v0.15.3](https://github.com/sile-typesetter/sile/releases/tag/v0.15.3) which fixed build issues and other bugs early in the major release cycle. This seem pretty stable now.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
 cd . && /bin/sh /private/tmp/sile-20240625-15901-jpiigt/sile-0.15.3/build-aux/missing automake-1.16 --foreign Makefile
/private/tmp/sile-20240625-15901-jpiigt/sile-0.15.3/build-aux/missing: line 81: automake-1.16: command not found
WARNING: 'automake-1.16' is missing on your system.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make: *** [Makefile.in] Error 127
```

**Comment by alerque** [comment]
@chenrui333 I've confirmed that is some upstream issue. For some reason even our GNU-Autotools generated source tarball is requiring automake at build time, which is objectively incorrect. After some dispute with the newish formula linting rules I've added a temporary workaround until a future upstream release fixes the root issue.

**Comment by alerque** [comment]
I've now tracked down the upstream bug and it will be fixed in the next release, but that might not be right away. Armed with the knowledge of exactly what happened I improved the workaround to make it even easier to back out later when a future release makes in unnecessary.

**Comment by alerque** [comment]
It is not stale and it does not need closing. It needs reviewing!

**Comment by alerque** [comment]
It is not stale and it does not need closing. It needs reviewing!

(Pro tip: If contributors can't do anything to move things forward then a bot shouldn't be pestering them here!)

**Comment by chenrui333** [comment]
it got buried in the prs (and also now we enforce the single commit per formula, otherwise it would show as error for github status), just did the commit squash, should be good now.

**Review by chenrui333 — APPROVED** [review]

File: Formula/s/sile.rb [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-  # - Switch `lua` dependency to `luajit` and clean up `lua` references
-  stable do
-    url "https://github.com/sile-typesetter/sile/releases/download/v0.14.17/sile-0.14.17.tar.xz"
-    sha256 "7f89bedecedabb5168250ad9dd80c09ed289c8e88c3d0d756d2d1d92ee065e04"
+  url "https://github.com/sile-typesetter/sile/releases/download/v0.15.3/sile-0.15.3.tar.zst"
+  sha256 "88d24a70710e301ba3400351a82c5f28b49cebcd637b7a59d28b001a56b398c6"
 
-    depends_on "lua"
-  end
+  url "https://github.com/sile-typesetter/sile.git", branch: "develop" if build.head?
```
Comment (chenrui333) [inline Formula/s/sile.rb:8]:
```suggestion
  url "https://github.com/sile-typesetter/sile.git", branch: "develop"
```

File: Formula/s/sile.rb [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
+
+  license "MIT"
```
Comment (chenrui333) [inline Formula/s/sile.rb:10]: should be moved before head

File: Formula/s/sile.rb (hunk 1 above)
Comment (alerque) [reply Formula/s/sile.rb]: This suggestion isn't correct, the `--HEAD` build has a different URl than the default. But we *could* drop the different branch bit, that is kind of obsolete now.

**Comment by chenrui333** [comment]
missing freetype for osx 

image

**Comment by alerque** [comment]
Um, the force push you just did @chenrui333 reverted the branch fix I'd pushed before and brought in a bunch of unrelated stuff.

**Comment by alerque** [comment]
The irrelevant stuff might have been rebase related, but you clobbered by branch fix. I just re posted that.

The freetype fix looks fine presuming macos doesn't have a version that could be used with `uses_from_macos`.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
> The freetype fix looks fine presuming macos doesn't have a version that could be used with `uses_from_macos`.

I dont think it applies in here? it is the indirect linkage only on macos?

**Comment by alerque** [comment]
> it is the indirect linkage only on macos?

Whatever kind of linkage ends up happening, freetype is only use at all on Mac OS. On Linux it isn't used at all (any more, it did used to be).

**Review by carlocab — APPROVED** [review]

**Pull request merged**